numpy/random/_generator.pyx

```
@@ -312,13 +311,13 @@ cdef class Generator:
 
         """
         cdef double temp
-        key = np.dtype(dtype).name
-        if key == 'float64':
+        _dtype = np.dtype(dtype)
```

Would normally spell this
```suggestion
        dtype = np.dtype(dtype)
```
Does cython not like that?

---

If I change it, test are failing... seems that it is not proper.

---

Do you have a link to the CI failure?

---

We need the original value so we can raise an error with it later. What about `descr` or `concrete_dtype`?